module/dice/dice.mjs

```diff
@@ -3,44 +3,56 @@
- * @param {object} [config.dialogOptions]    Modal dialog options
+ * ## Roll Configuration Dialog
+ * @property {boolean} [fastForward=false]     Should the roll configuration dialog be skipped?
+ * @property {boolean} [chooseModifier=false]  If the configuration dialog is shown, should the ability modifier be
+ *                                             configurable within that interface?
+ * @property {string} [template]               The HTML template used to display the roll configuration dialog.
+ * @property {string} [title]                  Title of the roll configuration dialog.
+ * @property {object} [dialogOptions]          Additional options passed to the roll configuration dialog.
  *
- * @param {boolean} [config.chatMessage=true] Automatically create a Chat Message for the result of this roll
- * @param {object} [config.messageData={}] Additional data which is applied to the created Chat Message, if any
- * @param {string} [config.rollMode]       A specific roll mode to apply as the default for the resulting roll
- * @param {object} [config.speaker]        The ChatMessage speaker to pass when creating the chat
- * @param {string} [config.flavor]         Flavor text to use in the posted chat message
+ * ## Chat Message
+ * @property {boolean} [chatMessage=true]  Should a chat message be created for this roll?
+ * @property {object} [messageData={}]     Additional data which is applied to the created chat message.
+ * @property {string} [rollMode]           Value of `CONST.DICE_ROLL_MODES` to apply as default for the chat message.
+ * @property {object} [flavor]             Flavor text to use in the created chat message.
+ */
+
+/**
+ * A standardized helper function for managing core 5e d20 rolls.
+ * Holding SHIFT, ALT, or CTRL when the attack is rolled will "fast-forward".
+ * This chooses the default options of a normal attack with no bonus, Advantage, or Disadvantage respectively
  *
- * @returns {Promise<D20Roll|null>}  The evaluated D20Roll, or null if the workflow was cancelled
+ * @param {D20RollConfiguration} configuration  Configuration data for the D20 roll.
+ * @returns {Promise<D20Roll|null>}             The evaluated D20Roll, or null if the workflow was cancelled.
  */
 export async function d20Roll({
-  parts=[], data={}, // Roll creation
-  advantage, disadvantage, fumble=1, critical=20, targetValue, elvenAccuracy, halflingLucky, reliableTalent, // Roll customization
-  chooseModifier=false, fastForward=false, event, template, title, dialogOptions, // Dialog configuration
-  chatMessage=true, messageData={}, rollMode, speaker, flavor // Chat Message customization
+  parts=[], data={}, event,
+  advantage, disadvantage, critical=20, fumble=1, targetValue,
+  elvenAccuracy, halflingLucky, reliableTalent,
+  fastForward=false, chooseModifier=false, template, title, dialogOptions,
+  chatMessage=true, messageData={}, rollMode, flavor,
+  speaker // Deprecated
```

Comment:
should we take this opportunity to nuke the deprecated speaker, or is that a job for another PR in 2.0.0?